src/pytestqt/qml/qmlbot.py

```diff
@@ -0,0 +1,38 @@
+import os
+from pathlib import Path
+from typing import Any
+
+from pytestqt.qt_compat import qt_api
+
+
+class QmlBot:
+    def __init__(self) -> None:
+        self.engine = qt_api.QtQml.QQmlApplicationEngine()
+        main = Path(__file__).parent / "botloader.qml"
+        self.engine.load(os.fspath(main))
+
+    @property
+    def _loader(self) -> Any:
+        self._root = self.engine.rootObjects()[
+            0
+        ]  # self is needed for it not to be collected by the gc
+        return self._root.findChild(qt_api.QtQuick.QQuickItem, "contentloader")
+
+    def loads(self, content: str) -> Any:
```

Comment:
FYI decided to drop the `load(path: Path) -> Any` variant, because the user can easily just call `loads(p.read_text(encoding=...))` instead of `load(p)`, with the advantage that we do not need to guess an encoding.

Reply:
no, that is not the same. files can import other qml files based on relative path.

Reply:
Not sure I follow, what does the fact that "qml files can import other qml files" relates to being able to load a qml file passing a `Path`? Can you exemplify?

Feel free to add it back, I removed because it seemed redundant and it the test was failing, but if it is not redundant we can revisit this.

Reply:
you could though pass the pass in the url here `self._comp.setData(content.encode("utf-8"), qt_api.QtCore.QUrl())`  I think 
though I don't know if it is really the same under the hood.

Reply:
In QML you can evrey file encapsulates a `Component` i.e if `MyApp.qml` lives near `main.qml`
You would do
```qml
import QtQuick

MyApp{}
```

Reply:
Ahh on CI it is showing the same error I got locally:

```
_____________________________ test_load_from_file _____________________________

qmlbot = <pytestqt.qml.qmlbot.QmlBot object at 0x00000193133134C8>

    def test_load_from_file(qmlbot: QmlBot) -> None:
        item = qmlbot.load(Path(__file__).parent / "sample.qml")
>       assert item.property("hello") == "world"
E       AttributeError: 'NoneType' object has no attribute 'property'
```

Reply:
are you on windows?

Reply:
Yes. 👍